dsn_test.go

```diff
@@ -387,14 +409,36 @@ func TestDSN(t *testing.T) {
 				Schema:             "sc",
 				Role:               "ro",
 				Region:             "b",
-				Authenticator:      "au",
+				Authenticator:      AuthTypeSnowflake,
 				Passcode:           "db",
 				PasscodeInPassword: true,
 				LoginTimeout:       10 * time.Second,
 				RequestTimeout:     300 * time.Second,
 				Application:        "special go",
 			},
-			dsn: "u:p@a.b.snowflakecomputing.com:443?application=special+go&authenticator=au&database=db&loginTimeout=10&passcode=db&passcodeInPassword=true&region=b&requestTimeout=300&role=ro&schema=sc",
+			dsn: "u:p@a.b.snowflakecomputing.com:443?application=special+go&database=db&loginTimeout=10&passcode=db&passcodeInPassword=true&region=b&requestTimeout=300&role=ro&schema=sc",
+		},
+		{
+			cfg: &Config{
+				User:               "u",
+				Password:           "p",
+				Account:            "a",
+				Authenticator:      AuthTypeExternalBrowser,
+			},
+			dsn: "u:p@a.snowflakecomputing.com:443?authenticator=externalbrowser",
+		},
+		{
+			cfg: &Config{
+				User:               "u",
+				Password:           "p",
+				Account:            "a",
+				Authenticator:      AuthTypeOkta,
+				OktaURL:   			&url.URL{
+					Scheme: "https",
+					Host: 	"sc.okta.com",
+				},
+			},
+			dsn: "u:p@a.snowflakecomputing.com:443?authenticator=https%253a%252f%252fsc.okta.com",
```

Comment:
This escaped URL doesn't looks right. It should be:
```
https%3A%2F%2Fsc.okta.com
```
I guess you are double escaping the URL parameter. See the above comment.

Reply:
Thanks for pointing it out. I just remove the double escaping